services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteOptions.java

```diff
@@ -1871,4 +1979,313 @@ public Builder snappingIncludeClosuresList(@Nullable List<Boolean> snappingClosu
     @NonNull
     public abstract RouteOptions build();
   }
+
+  /**
+   * Object representing experimental value.
+   * <p>
+   * All available experimental values are subject to change at any time.
+   */
+  @AutoValue
+  public abstract static class Experimental extends DirectionsJsonObject {
+
+    static String[] experimentalParameters = new String[] {
```

Comment:
It seems like the `experimentalParameters` array is used only for lookup if a filed is experimental. Maybe it's easier to keep all the experimental fields name in a `HashSet`?

Reply:
@VysotskiVadim what would be a practical difference?

Reply:
You won't need to implement custom lookup with HashSet. Instead of 
```kotlin
        String property = URLDecoder.decode(query.substring(0, idx), UTF_8);
        String value = URLDecoder.decode(query.substring(idx + 1), UTF_8);

        boolean isExperimental = false;
        for (int i = 0; i < Experimental.experimentalParameters.length; i++) {
          if (property.equals(Experimental.experimentalParameters[i])) {
            isExperimental = true;
            if (!optionsJson.has("experimental")) {
              optionsJson.add("experimental", new JsonObject());
            }
            JsonObject experimental = optionsJson.getAsJsonObject("experimental");
            experimental.addProperty(property, value);
            break;
          }
        }
```

you will be able to do:
```kotin
        String property = URLDecoder.decode(query.substring(0, idx), UTF_8);
        String value = URLDecoder.decode(query.substring(idx + 1), UTF_8);

        boolean isExperimental = false;
        if (experimentalParameters.contains(property)) {
          isExperimental = true;
          if (!optionsJson.has("experimental")) {
            optionsJson.add("experimental", new JsonObject());
          }
          JsonObject experimental = optionsJson.getAsJsonObject("experimental");
          experimental.addProperty(property, value);
        }
        
```

Also runtime complexity of lookup in hash table is O(1) vs O(n) in array. But I don't think performance is a good reason for this change, because array has small and constant size.